File: BinaryTree/csharp/TreeValueCount.cs
namespace BinaryTreeSolutions;

public static class TreeValueCount
{
    public static int Solve(TreeNode? root, int target)
    {
        if (root is null)
        {
            return 0;
        }

        var match = root.Val == target ? 1 : 0;
        return match + Solve(root.Left, target) + Solve(root.Right, target);
    }
}

File: BinaryTree/php/treeValueCount.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/TreeNode.php';

function treeValueCount(?TreeNode $root, int $target): int
{
    if ($root === null) {
        return 0;
    }

    $match = $root->val === $target ? 1 : 0;
    return $match + treeValueCount($root->left, $target) + treeValueCount($root->right, $target);
}
